### ISSUE TITLE
Allow resizing of tiled windows

### ISSUE THREAD

**Issue by dakyskye** [avahe-kellenberger/nimdow#143]
I'd like to be able to resize Tiled windows (so I have increasing the size of Master window). It'd also be cool to add pseudo-tiled mode for a Window like bspwm does (so you resize Certain tiled window without resizing Other tiled windows while when a window is not pseudo-tiled and you resize it, it also causes other windows to resize).

**Comment by avahe-kellenberger** [comment]
Been wanting to implement this for a while.

My main questions are, after resizing a slave window:

1. What should happen if you spawn a new slave window? (Should the resize just be undone)
2. If you move the resized slave window, should that window stay the same size?

I'm thinking:
1. Undo the resizing
2. Stay the same size unless it can take up _more_ vertical space

**Comment by dakyskye** [comment]
Well, if we are not talking about pseude tiled windows, then:
You should be able to resize any window whether it's master or slave. Resizing a window will cause resizing of other windows around it. If you spawn a new slave window, its horizontal size should be the same as other slave windows, but for vertical size, it should try to be as big as possible in given space. When you move a slave window, it should remain the size it has, since the reason why certain slave window was resized was to see more output from a command and when I move that window I definitely want to keep the size of it.

**Comment by avahe-kellenberger** [comment]
Cool, sounds like we're on the same page.

I'm not sure what you mean by "pseudo-tiled" but I could implement this with the current layout system

**Comment by dakyskye** [comment]
Pseudo tiled means a window's size can't be increased, but it can be Decreased. And the difference is that when a slave window is not in pseudo tiled mode (so when it is in tiled mode), decreasing or increasing its size causes resizing of other windows, but in pseudo tiled's case, windows' size can get shrunk but other windows won't change their size. I will try to find a screenshot showing that. [bspwm](https://github.com/baskerville/bspwm) has this feature.

**Comment by uninhm** [comment]
![image](https://user-images.githubusercontent.com/62044675/120081762-9a850100-c095-11eb-9775-4329e5dd1ef6.png)

**Comment by uninhm** [comment]
If you find that useful I can do a pull request

**Comment by dakyskye** [comment]
Yeah, it would be great. So then only pseudo tiling and vertical resizing will be left to be implemented 🚀
